scripts/asm-installer/asm_vm

```diff
@@ -913,12 +913,10 @@ is_cluster_registered() {
 }
 
 populate_environ_info() {
-  if [[ -n "${ENVIRON_PROJECT_ID}" ]]; then return; fi
+  if [[ -n "${ENVIRON_PROJECT_ID}" && -n "${HUB_MEMBERSHIP_ID}" ]]; then return; fi
   if ! is_membership_crd_installed; then return; fi
   configure_kubectl
-  local IDENTITY_PROVIDER
-  IDENTITY_PROVIDER="$(kubectl get memberships.hub.gke.io membership -o=json | jq .spec.identity_provider)"
-  ENVIRON_PROJECT_ID="$(echo "${IDENTITY_PROVIDER}" | sed 's/^\"https:\/\/gkehub.googleapis.com\/projects\/\(.*\)\/locations\/global\/memberships\/\(.*\)\"$/\1/g')"
+  ENVIRON_PROJECT_ID="$(kubectl get memberships.hub.gke.io membership -o=json | jq .spec.workload_identity_pool | sed 's/^\"\(.*\).\(svc\|hub\).id.goog\"$/\1/g')"
```

Comment:
Just to confirm, this command returns the hub IDNS, is environ project id equal to IDNS? I thought environ project id is a project id.

Reply:
I see, you are correct. This extract project id from IDNS.